pymc_marketing/clv/plotting.py

```diff
@@ -0,0 +1,163 @@
+    Z = (
+        model.expected_num_purchases(
+            customer_id=np.arange(mesh_recency.size),  # placeholder
+            t=t,
+            frequency=mesh_frequency.ravel(),
+            recency=mesh_recency.ravel(),
+            T=max_recency,
+        )
+        .mean(("draw", "chain"))
+        .values.reshape(mesh_recency.shape)
+    )
+
+    ax = plt.subplot(111)
+    pcm = ax.imshow(Z, **kwargs)
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    if title is None:
+        title = (
+            "Expected Number of Future Purchases for {} Unit{} of Time,".format(
+                t, "s"[t == 1 :]
+            )
+            + "\nby Frequency and Recency of a Customer"
+        )
+    plt.title(title)
+
+    forceAspect(ax)
+
+    # plot colorbar beside matrix
+    plt.colorbar(pcm, ax=ax)
+
+    return ax
+
+
+def plot_probability_alive_matrix(
+    model,
+    max_frequency=None,
+    max_recency=None,
+    title="Probability Customer is Alive,\nby Frequency and Recency of a Customer",
+    xlabel="Customer's Historical Frequency",
+    ylabel="Customer's Recency",
+    **kwargs
+):
+    """
+    Plot probability alive matrix as heatmap.
+    Plot a figure of the probability a customer is alive based on their
+    frequency and recency.
+    Parameters
+    ----------
+    model: lifetimes model
+        A fitted lifetimes model.
+    max_frequency: int, optional
+        The maximum frequency to plot. Default is max observed frequency.
+    max_recency: int, optional
+        The maximum recency to plot. This also determines the age of the customer.
+        Default to max observed age.
+    title: str, optional
+        Figure title
+    xlabel: str, optional
+        Figure xlabel
+    ylabel: str, optional
+        Figure ylabel
+    kwargs
+        Passed into the matplotlib.imshow command.
+    Returns
+    -------
+    axes: matplotlib.AxesSubplot
+    """
+
+    if max_frequency is None:
+        max_frequency = int(model.frequency.max())
+
+    if max_recency is None:
+        max_recency = int(model.recency.max())
+
+    frequency = np.arange(max_frequency + 1)
+    recency = np.arange(max_recency + 1)
+    mesh_frequency, mesh_recency = np.meshgrid(frequency, recency)
+    Z = (
+        model.expected_probability_alive(
+            customer_id=np.arange(mesh_recency.size),  # placeholder
+            frequency=mesh_frequency.ravel(),
+            recency=mesh_recency.ravel(),
+            T=max_recency,
+        )
+        .mean(("draw", "chain"))
+        .values.reshape(mesh_recency.shape)
+    )
+
+    interpolation = kwargs.pop("interpolation", "none")
+
+    ax = plt.subplot(111)
+    pcm = ax.imshow(Z, interpolation=interpolation, **kwargs)
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    plt.title(title)
+
+    forceAspect(ax)
+
+    # plot colorbar beside matrix
+    plt.colorbar(pcm, ax=ax)
+
+    return ax
+
+
+def forceAspect(ax, aspect=1):
```

Comment:
shall we use `force_aspect` syntax?

Reply:
You mean without the `forceAspect`? We could remove it, but I believe that the shape is more rectangular.

> shall we use `force_aspect` syntax?

On the same note, the function was taken from [lifetimes/plotting.py](https://github.com/CamDavidsonPilon/lifetimes/blob/master/lifetimes/plotting.py#L674-L677). We could use `force_aspect` as a name, I don't have strong preferences. Should we include the lifetimes license since a lot of the `plotting.py` file was taken from the package?

Reply:
I think we can be conservative and add the necessary licenses. My comment was more about the function naming convention.

Reply:
Ahh, yes let's use the underscore convention. I thought you were referring to some functionality directly in matplotlib